### PR TITLE
PyTorch FastMRI Implementation

### PR DESCRIPTION
**Pull request by pomonam**
This PR implements U-Net on PyTorch for the FastMRI module.

- Adds the FastMRI workload and reference submission.
- Contains PyTorch U-Net and the FastMRI data loading implementations primarily adapted from https://github.com/facebookresearch/fastMRI. 

After the Jax implementation, the follow-up PR should:
- Implement test functions on `tests` folder.
- Compare consistency between Jax and PyTorch implementation.
- Finetune default hyperparameters and set the target value.

### CODE REVIEW

**Review by runame — APPROVED** [review]
LGTM! Have you already tried to run the code and if yes, how long did it take to reach the target performance? If you don't have access to sufficient resources to run it (ideally 8xV100s), just let me know, then I can try to run it.

File: algorithmic_efficiency/workloads/fastmri/fastmri_pytorch/workload.py [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+from algorithmic_efficiency.workloads.fastmri.fastmri_pytorch.input_pipeline import \
+    SliceDataset
+from algorithmic_efficiency.workloads.fastmri.fastmri_pytorch.input_pipeline import \
+    UnetDataTransform
+from algorithmic_efficiency.workloads.fastmri.fastmri_pytorch.models import \
+    unet
+from algorithmic_efficiency.workloads.fastmri.workload import \
+    BaseFastMRIWorkload
+
+PYTORCH_DDP = 'LOCAL_RANK' in os.environ
```
[the remark this answers is not in the document]
Comment (pomonam) [reply algorithmic_efficiency/workloads/fastmri/fastmri_pytorch/workload.py]: Fixed!

File: algorithmic_efficiency/workloads/fastmri/fastmri_pytorch/input_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+
+import contextlib
+import os
+from pathlib import Path
+import pickle
+import random
+from typing import Any, Callable, Dict, Optional, Sequence, Tuple, Union
+import xml.etree.ElementTree as etree
+
+import h5py
```
[the remark this answers is not in the document]
Comment (pomonam) [reply algorithmic_efficiency/workloads/fastmri/fastmri_pytorch/input_pipeline.py:15]: Fixed! I modified `setup.cfg` accordingly.

File: algorithmic_efficiency/workloads/fastmri/fastmri_pytorch/workload.py [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+  worker_info = torch.utils.data.get_worker_info()
+  data = worker_info.dataset
+
+  # for NumPy random seed we need it to be in this range
+  base_seed = worker_info.seed
+
+  if data.transform.mask_func is not None:
+    # DDP training: unique seed is determined by worker and device
+    if PYTORCH_DDP:
+      seed = base_seed + torch.distributed.get_rank() * worker_info.num_workers
```
[the remark this answers is not in the document]
Comment (pomonam) [reply algorithmic_efficiency/workloads/fastmri/fastmri_pytorch/workload.py]: Fixed!

File: algorithmic_efficiency/workloads/fastmri/fastmri_pytorch/workload.py [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+        sampler = data_utils.DistributedEvalSampler(
+            dataset, num_replicas=N_GPUS, rank=RANK, shuffle=False)
+      batch_size //= N_GPUS
+    dataloader = torch.utils.data.DataLoader(
+        dataset,
+        batch_size=batch_size,
+        shuffle=not PYTORCH_DDP and is_train,
+        worker_init_fn=worker_init_fn,
+        sampler=sampler,
+        num_workers=0,
```
[the remark this answers is not in the document]
Comment (pomonam) [reply algorithmic_efficiency/workloads/fastmri/fastmri_pytorch/workload.py]: I temporarily increased `num_workers` to `4`.

**Comment by znado** [comment]
Looks great, thanks so much for the quick implementation! Going to merge now, and if we hit issues running on the 8x V100s we can fix forward.

**Review by znado — APPROVED** [review]

**Pull request merged**